source/opt/instrument_pass.h

```diff
@@ -76,7 +76,7 @@ class InstrumentPass : public Pass {
            IRContext::kAnalysisInstrToBlockMapping |
            IRContext::kAnalysisDecorations | IRContext::kAnalysisCombinators |
            IRContext::kAnalysisNameMap | IRContext::kAnalysisBuiltinVarId |
-           IRContext::kAnalysisConstants | IRContext::kAnalysisTypes;
+           IRContext::kAnalysisConstants;
```

Comment:
I'm curious how the type manager gets out of date.  You use the type manager a lot during the pass.  Are you sure that it is still valid enough during the pass that it will not give you incorrect results?

Reply:
It is out of date because of the same issue we have been discussing above: it does not account for the decorations I have added to two different types I created.